medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -94,19 +95,37 @@ pub async fn process_message(message: DeleteMessage) -> Result<Uuid, ProcessMess
         &blob_name,
         &message.document_content_id
     );
+
+    match delete_blob(&storage_client, &storage_container_name, &blob_name).await {
+        Ok(_) => {
+            tracing::info!(
+                "Deleted blob {} from storage container {}",
+                &blob_name,
+                &storage_container_name,
+            );
+            Ok(())
+        }
+        Err(e) => match e {
+            AzureError::UnexpectedHTTPResult(e) => {
+                if e.status_code() == StatusCode::NOT_FOUND {
+                    tracing::warn!(
+                        "Blob {} couldn't be found in storage container {}: {:?}",
+                        &blob_name,
+                        &storage_container_name,
+                        e
+                    );
+                    Ok(())
+                } else {
+                    tracing::error!("Error deleting blob: {:?}", e);
+                    Err(anyhow!("Couldn't delete blob {}: {:?}", &blob_name, e))
+                }
+            }
+            e => Err(anyhow!("Couldn't delete blob {}: {:?}", &blob_name, e)),
+        },
+    }?;
```

Comment:
All of this error handling should be moved into `delete_blob` and brought under test, but doing that would take me longer than we have left in today.

Reply:
Does this correctly set the job status to an error if the blob couldn't be found?

Reply:
No. It assumes that a blob not being found is OK.

I don't think the blob not being found in blob storage when it is found in the search index is an error.

Reply:
@m-doughty yes it looks like it would. This was the previous behaviour, when the error was mapped to the `anyhow!("Couldn't delete blob {}", &blob_name)` error - the status updated to this.

This looks ok as a stop-gap fix to me, with a clean-up to follow.

Reply:
> No. It assumes that a blob not being found is OK.
> 
> I don't think the blob not being found in blob storage when it is found in the search index is an error.

This discussion was had with Accenture, they are requesting an error state when the document they attempted to delete wasn't found in Blob storage.